File: scala-pr/54/scala/code-quality-logging/RequestDebugger.scala
case class Request(url: String, headers: Map[String, String], body: String, session: Map[String, Any])

class RequestDebugger {
  def debugRequest(request: Request): Unit = {
    // {fact rule=code-quality-logging@v1.0 defects=1}
    println(s"Request URL: ${request.url}")
    println(s"Request headers: ${request.headers}")
    println(s"Authorization header: ${request.headers.get("Authorization")}")
    println(s"Request body: ${request.body}")
    println(s"Session data: ${request.session}")
    // {/fact}
  }
}

File: kotlin-pr/50/kotlin/code-quality-logging/RequestDebugger.kt
data class Request(
    val url: String,
    val headers: Map<String, String>,
    val body: String,
    val session: Map<String, Any>
)

class RequestDebugger {
    fun debugRequest(request: Request) {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        println("Request URL: ${request.url}")
        println("Request headers: ${request.headers}")
        println("Authorization header: ${request.headers["Authorization"]}")
        println("Request body: ${request.body}")
        println("Session data: ${request.session}")
        // {/fact}
    }
}